### PR TITLE
Most popular food endpoint

### PR DESCRIPTION
**Pull request by JoriPeterson**
Adds endpoint and test for most_popular_food endpoint

### CODE REVIEW

File: routes/api/v1/meals.js [hunk 1]
```diff
@@ -62,4 +71,28 @@ router.delete('/:meal_id/foods/:id', (req, res) => {
...
+      where: {
+        id: mostPopular[0].FoodId
+      }
+    })
+    .then(food => {
+      res.status(200).send(food)
+    })
+  })
+  .catch(error => res.status(500).send({error}));
+})
```
Comment (Myrdden) [inline routes/api/v1/meals.js]:
```suggestion
router.get('/most_popular_food', async (req, res) => {
  res.setHeader('Content-Type', 'application/json');
  
  let counted_foods = await MealFood.findAll({
    attributes:
      ['FoodId', [Sequelize.fn('count', Sequelize.col('FoodId')), 'count']],
      group: ['MealFoods.FoodId'],
      raw: true,
      order: sequelize.literal('count DESC LIMIT 1')
  });
  
  res.status(200).send(counted_foods[0]);
});
```

Does this work?

Comment (JoriPeterson) [reply routes/api/v1/meals.js]: This doesn't connect to the Food table for name and calories

Comment (Myrdden) [reply routes/api/v1/meals.js]: Heck
```suggestion
router.get('/most_popular_food', async (req, res) => {
  res.setHeader('Content-Type', 'application/json');
  
  let countedFoods = await MealFood.findAll({
    attributes:
    ['FoodId', [Sequelize.fn('count', Sequelize.col('FoodId')), 'count']],
    group: ['MealFoods.FoodId'],
    raw: true,
    order: sequelize.literal('count DESC LIMIT 1')
  });
  let mostPopular = await Food.findOne({where: {id: countedFoods[0].FoodId}});
  res.status(200).send(mostPopular);
});
```
???

Comment (JoriPeterson) [reply routes/api/v1/meals.js]: I tested it and this works. Same thing as before just with async/await instead of promises

**Review by Myrdden — APPROVED** [review]

**Pull request merged**